packages/framework/src/hooks/useScreenData.ts

```diff
@@ -0,0 +1,58 @@
+import { useAtom, useAtomValue } from "jotai";
+import { clone } from "lodash-es";
+import { useCallback } from "react";
+import isEqual from "react-fast-compare";
+import type { Response, WebSocketConnection } from "../data";
+import type {
+  EnsembleAPIModel,
+  EnsembleMockResponse,
+  EnsembleSocketModel,
+} from "../shared";
+import type { ScreenContextDefinition, ScreenContextActions } from "../state";
+import { screenApiAtom, screenSocketAtom, screenDataAtom } from "../state";
+import { useEvaluate } from "./useEvaluate";
+
+export const useScreenData = (): { apis?: EnsembleAPIModel[] } & {
+  sockets?: EnsembleSocketModel[];
+} & Pick<ScreenContextDefinition, "data"> &
+  Pick<ScreenContextActions, "setData"> & {
+    mockResponses: { [key: string]: EnsembleMockResponse | string | undefined };
+  } => {
```

Comment:
Sorry this isn't your mess but can you help clean up the typing on this and create a new type to capture the return value?